src/PanelTraits/AutoSet.php

```diff
@@ -52,7 +53,7 @@ public function getDbColumnTypes()
         foreach ($table_columns as $key => $column) {
             $column_type = $this->model->getConnection()->getSchemaBuilder()->getColumnType($this->model->getTable(), $column);
             $this->db_column_types[$column]['type'] = trim(preg_replace('/\(\d+\)(.*)/i', '', $column_type));
-            $this->db_column_types[$column]['default'] = ''; // no way to do this using DBAL?!
+            $this->db_column_types[$column]['default'] = $this->model->getConnection()->getDoctrineSchemaManager()->listTableDetails($this->model->getTable())->getColumn($column)->getDefault();
```

Comment:
In debated doing something like that. In that same vein, would this be preferred?

```php
    public function getDbColumnTypes()
    {
        $table = $this->model->getTable();
        $conn = $this->model->getConnect();
        $table_columns = $conn->getSchemaBuilder()->getColumnListing($table);

        foreach ($table_columns as $key => $column) {
            $column_type = $conn->getSchemaBuilder()->getColumnType($table, $column);
            $this->db_column_types[$column]['type'] = trim(preg_replace('/\(\d+\)(.*)/i', '', $column_type));
            $this->db_column_types[$column]['default'] = $conn->getDoctrineSchemaManager()->listTableDetails($table)->getColumn($column)->getDefault();
        }
        return $this->db_column_types;
    }
```